src/commands/list.js

```diff
@@ -50,71 +48,65 @@ class ListCommand extends AbstractCommand {
 
     this.logger.warn('Querying cloud accounts, regions and services. It might take a while...');
 
-    return this._getCredentials()
-      .then(credentials => {
-        const sts = new AWS.STS({ credentials });
-
-        return sts.getCallerIdentity().promise();
-      })
-      .then(identity => {
-        this.accountId = identity.Account;
-
-        return this._fetchResources();
-      })
-      .then(data => {
-        /**
-         * @param {Array} allowed
-         * @param {String} item
-         * @returns {Boolean}
-         */
-        function isAllowed(allowed, item) {
-          return (allowed.length === 0 || allowed.includes(item));
-        }
+    try {
+      const credentials = await this._getCredentials();
 
-        this.logger.log('Compiling the list of cloud resources. ' +
-          'Use --depth, -d option to view details about projects, accounts, regions and services.' + os.EOL);
-
-        data.forEach(item => {
-          if (
-            isAllowed(projects, item.project)
-            && isAllowed(accounts, item.accountId)
-            && isAllowed(regions, item.region)
-            && isAllowed(services, item.service)
-          ) {
-            this.hash.set([item.project, item.accountId, item.region, item.service, item.resource].join(':'), null);
-          }
-        });
-
-        return Promise.resolve();
-      })
-      .then(() => {
-        const projectsData = this.hash.getRaw();
-
-        if (Object.keys(projectsData).length) {
-          this.logger.log('Projects');
-
-          this._showTree(this._format(projectsData, 0, depth));
-        } else {
-          this.logger.log('No Projects to display.');
-        }
+      const sts = new AWS.STS({ credentials });
+      const identity = await sts.getCallerIdentity().promise();
```

Comment:
@todo in future, there must be a fall back if user is NOT allowed to call `STS` with action `getCallerIdentity`